### DISCUSSION TITLE
How to set single precision for writing plotfile, while keeping double precision for checkpoint file?

### DISCUSSION THREAD

**Discussion by smahesh14** [AMReX-Codes/amrex#2625] [Q&A] [answered]
I am using PRECISION = DOUBLE while compiling. I want to write the plotfiles in single precision for which in the input file I set :

fab.format = NATIVE_32

But this writes the checkpoint file in single precision which is not desired. How can I write checkpoint file in double precision and the plotfile in single precision?

**Comment by cgilet** [comment, marked as answer]
`AmrCore` doesn't provide a specific function for writing checkpoint files; it must exist in your application code. However, you can take a look at the function `Amr::checkPoint` found in amrex/Src/Amr/AMReX_Amr.cpp, (which always writes checkpoint files in the "NATIVE" format, which would be double precision for a code compiled with PRECISION = DOUBLE) to see how you might edit the application code to set the checkpoint file precision separately. ... Either that or open an issue on the github page of the application code.

**Reply by smahesh14** [reply]
Thanks for the quick reply.

The specific lines of code (from Amr::checkPoint in amrex/Src/Amr/AMReX_Amr.cpp) that can set the NATIVE format and reset it to previously set format :

    FABio::Format thePrevFormat = FArrayBox::getFormat();
    FArrayBox::setFormat(FABio::FAB_NATIVE);
   
   // .... 

    FArrayBox::setFormat(thePrevFormat);
